src/apm_client.rs

```diff
@@ -73,12 +75,20 @@ impl ApmClient {
                 }
             })
             .map(Arc::new);
-        let client = if allow_invalid_certs {
-            let builder = reqwest::ClientBuilder::new().danger_accept_invalid_certs(true);
-            builder.build().unwrap()
-        } else {
-            Client::new()
-        };
+
+        let mut client_builder = reqwest::ClientBuilder::new();
+        if allow_invalid_certs {
+            client_builder = client_builder.danger_accept_invalid_certs(true);
+        }
+        if let Some(path) = root_cert_path {
+            let mut buff = Vec::new();
+            std::fs::File::open(&path).unwrap()
```

Comment:
It would be better to return a Result, instead of unwrapping here and later below. We shouldn't panic on recoverable errors.